src/lib/hash-parser-hoc.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+/* Higher Order Component to get the project id from location.hash
+ * @param {React.Component} WrappedComponent component to receive projectData prop
+ * @returns {React.Component} component with project loading behavior
+ */
+const HashParserHOC = function (WrappedComponent) {
+    class HashParserComponent extends React.Component {
+        constructor (props) {
+            super(props);
+            this.fetchProjectId = this.fetchProjectId.bind(this);
+            this.updateProject = this.updateProject.bind(this);
+            this.state = {
+                projectId: null
+            };
+        }
+        componentDidMount () {
+            window.addEventListener('hashchange', this.updateProject);
```

Comment:
Seeing it in this new context, I think a better name for `updateProject` here would be `handleHashChange`.